stability: data races on rho

build sha: 1a34ca3

Multiple data races detected on rho:
`104.196.102.49`:
```
==================
WARNING: DATA RACE
Write at 0x00c429930f30 by goroutine 247:
  runtime.mapassign1()
      /usr/local/go/src/runtime/hashmap.go:442 +0x0
  github.com/cockroachdb/cockroach/storage.(*Store).Send.func1()
      /go/src/github.com/cockroachdb/cockroach/storage/store.go:1970 +0x2a2
  github.com/cockroachdb/cockroach/storage.(*Store).Send()
      /go/src/github.com/cockroachdb/cockroach/storage/store.go:2105 +0xcb6
  github.com/cockroachdb/cockroach/storage.(*Stores).Send()
      /go/src/github.com/cockroachdb/cockroach/storage/stores.go:182 +0x24a
  github.com/cockroachdb/cockroach/server.(*Node).Batch.func3()
      /go/src/github.com/cockroachdb/cockroach/server/node.go:818 +0x3a8
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunTask()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:215 +0x10d
  github.com/cockroachdb/cockroach/server.(*Node).Batch()
      /go/src/github.com/cockroachdb/cockroach/server/node.go:830 +0x2ce
  github.com/cockroachdb/cockroach/roachpb._Internal_Batch_Handler()
      /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:1497 +0x345
  google.golang.org/grpc.(*Server).processUnaryRPC()
      /go/src/google.golang.org/grpc/server.go:611 +0x10b6
  google.golang.org/grpc.(*Server).handleStream()
      /go/src/google.golang.org/grpc/server.go:769 +0xf9c
  google.golang.org/grpc.(*Server).serveStreams.func1.1()
      /go/src/google.golang.org/grpc/server.go:422 +0xb8

Previous read at 0x00c429930f30 by goroutine 95:
  [failed to restore the stack]

Goroutine 247 (running) created at:
  google.golang.org/grpc.(*Server).serveStreams.func1()
      /go/src/google.golang.org/grpc/server.go:423 +0xba
  google.golang.org/grpc/transport.(*http2Server).operateHeaders()
      /go/src/google.golang.org/grpc/transport/http2_server.go:221 +0x126d
  google.golang.org/grpc/transport.(*http2Server).HandleStreams()
      /go/src/google.golang.org/grpc/transport/http2_server.go:282 +0x7f3
  google.golang.org/grpc.(*Server).serveStreams()
      /go/src/google.golang.org/grpc/server.go:424 +0x1d5
  google.golang.org/grpc.(*Server).serveNewHTTP2Transport()
      /go/src/google.golang.org/grpc/server.go:411 +0x4f0
  google.golang.org/grpc.(*Server).handleRawConn()
      /go/src/google.golang.org/grpc/server.go:388 +0x5c2

Goroutine 95 (running) created at:
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x73
  github.com/cockroachdb/cockroach/kv.grpcTransportFactory()
      /go/src/github.com/cockroachdb/cockroach/kv/transport_race.go:100 +0x2d8
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:981 +0x157
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:364 +0x27b
  github.com/cockroachdb/cockroach/kv.(*DistSender).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:250 +0x31d
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:459 +0x4da
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:344 +0x677
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:282 +0xbf
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:455 +0x6c8
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:344 +0x677
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:282 +0xbf
  github.com/cockroachdb/cockroach/kv.(*DistSender).getDescriptors()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:429 +0x101
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendChunk()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:672 +0x3d3
  github.com/cockroachdb/cockroach/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:591 +0x1a0
  github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).Send()
      /go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:383 +0x641
  github.com/cockroachdb/cockroach/internal/client.(*txnSender).Send()
      /go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:52 +0x120
  github.com/cockroachdb/cockroach/internal/client.(*DB).send()
      /go/src/github.com/cockroachdb/cockroach/internal/client/db.go:490 +0x2be
  github.com/cockroachdb/cockroach/internal/client.(*Txn).send()
      /go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:675 +0x675
  github.com/cockroachdb/cockroach/internal/client.(*Txn).(github.com/cockroachdb/cockroach/internal/client.send)-fm()
      /go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:359 +0x7c
  github.com/cockroachdb/cockroach/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/internal/client/db.go:417 +0x1ab
  github.com/cockroachdb/cockroach/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:359 +0xfd
  github.com/cockroachdb/cockroach/internal/client.runOneResult()
      /go/src/github.com/cockroachdb/cockroach/internal/client/db.go:517 +0x97
  github.com/cockroachdb/cockroach/internal/client.runOneRow()
      /go/src/github.com/cockroachdb/cockroach/internal/client/db.go:531 +0x6d
  github.com/cockroachdb/cockroach/internal/client.(*Txn).Get()
      /go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:221 +0xfa
  github.com/cockroachdb/cockroach/sql.(*LeaseManager).resolveName()
      /go/src/github.com/cockroachdb/cockroach/sql/lease.go:1061 +0xd1
  github.com/cockroachdb/cockroach/sql.(*LeaseManager).AcquireByName()
      /go/src/github.com/cockroachdb/cockroach/sql/lease.go:995 +0x138
  github.com/cockroachdb/cockroach/sql.(*planner).getTableLease()
      /go/src/github.com/cockroachdb/cockroach/sql/table.go:214 +0x4db
  github.com/cockroachdb/cockroach/sql.(*planner).(github.com/cockroachdb/cockroach/sql.getTableLease)-fm()
      /go/src/github.com/cockroachdb/cockroach/sql/scan.go:251 +0x4f
  github.com/cockroachdb/cockroach/sql.(*scanNode).initTable()
      /go/src/github.com/cockroachdb/cockroach/sql/scan.go:259 +0xe6
  github.com/cockroachdb/cockroach/sql.(*planner).getDataSource()
      /go/src/github.com/cockroachdb/cockroach/sql/data_source.go:243 +0x146e
  github.com/cockroachdb/cockroach/sql.(*planner).getDataSource()
      /go/src/github.com/cockroachdb/cockroach/sql/data_source.go:280 +0x178
  github.com/cockroachdb/cockroach/sql.(*planner).getSources()
      /go/src/github.com/cockroachdb/cockroach/sql/data_source.go:177 +0xae
  github.com/cockroachdb/cockroach/sql.(*selectNode).initFrom()
    /go/src/github.com/cockroachdb/cockroach/sql/data_source.go:177 +0xae
  github.com/cockroachdb/cockroach/sql.(*selectNode).initFrom()
      /go/src/github.com/cockroachdb/cockroach/sql/select.go:438 +0x112
  github.com/cockroachdb/cockroach/sql.(*planner).SelectClause()
      /go/src/github.com/cockroachdb/cockroach/sql/select.go:295 +0x158
  github.com/cockroachdb/cockroach/sql.(*planner).Select()
      /go/src/github.com/cockroachdb/cockroach/sql/select.go:244 +0xa85
  github.com/cockroachdb/cockroach/sql.(*planner).prepare()
      /go/src/github.com/cockroachdb/cockroach/sql/plan.go:321 +0xe9b
  github.com/cockroachdb/cockroach/sql.(*Executor).Prepare()
      /go/src/github.com/cockroachdb/cockroach/sql/executor.go:366 +0x8c6
  github.com/cockroachdb/cockroach/sql.PreparedStatements.New()
      /go/src/github.com/cockroachdb/cockroach/sql/prepare.go:72 +0x84
  github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).handleParse()
      /go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:367 +0x689
  github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).serve()
      /go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:282 +0x8d1
  github.com/cockroachdb/cockroach/sql/pgwire.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/sql/pgwire/server.go:240 +0x989
  github.com/cockroachdb/cockroach/server.(*Server).Start.func8.1()
      /go/src/github.com/cockroachdb/cockroach/server/server.go:391 +0x88
  github.com/cockroachdb/cockroach/util/netutil.(*Server).ServeWith.func1()
      /go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:138 +0xc8
==================
```

Probably the same (`github.com/cockroachdb/cockroach/storage.(*Store).Send.func1()` is involved in both):
```
==================
WARNING: DATA RACE
Read at 0x00c42b517980 by goroutine 95:
  reflect.maplen()
      /usr/local/go/src/runtime/hashmap.go:1044 +0x0
  reflect.Value.MapKeys()
      /usr/local/go/src/reflect/value.go:1072 +0x427
  encoding/gob.(*Encoder).encodeMap()
      /usr/local/go/src/encoding/gob/encode.go:377 +0x225
  encoding/gob.encOpFor.func3()
      /usr/local/go/src/encoding/gob/encode.go:577 +0x158
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeArray()
      /usr/local/go/src/encoding/gob/encode.go:357 +0x2c9
  encoding/gob.encOpFor.func1()
      /usr/local/go/src/encoding/gob/encode.go:557 +0x170
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.(*Encoder).encode()
      /usr/local/go/src/encoding/gob/encode.go:707 +0x257
  encoding/gob.(*Encoder).EncodeValue()
      /usr/local/go/src/encoding/gob/encoder.go:247 +0x6ea
  encoding/gob.(*Encoder).Encode()
      /usr/local/go/src/encoding/gob/encoder.go:175 +0x72
  github.com/cockroachdb/cockroach/kv.grpcTransportFactory.func1()
      /go/src/github.com/cockroachdb/cockroach/kv/transport_race.go:94 +0x39b
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x8a

Previous write at 0x00c42b517980 by goroutine 333:
  runtime.mapassign1()
      /usr/local/go/src/runtime/hashmap.go:442 +0x0
  github.com/cockroachdb/cockroach/storage.(*Store).Send.func1()
      /go/src/github.com/cockroachdb/cockroach/storage/store.go:1970 +0x2a2
  github.com/cockroachdb/cockroach/storage.(*Store).Send()
      /go/src/github.com/cockroachdb/cockroach/storage/store.go:2105 +0xcb6
  github.com/cockroachdb/cockroach/storage.(*Stores).Send()
      /go/src/github.com/cockroachdb/cockroach/storage/stores.go:182 +0x24a
  github.com/cockroachdb/cockroach/server.(*Node).Batch.func3()
      /go/src/github.com/cockroachdb/cockroach/server/node.go:818 +0x3a8
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunTask()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:215 +0x10d
  github.com/cockroachdb/cockroach/server.(*Node).Batch()
      /go/src/github.com/cockroachdb/cockroach/server/node.go:830 +0x2ce
  github.com/cockroachdb/cockroach/roachpb._Internal_Batch_Handler()
      /go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:1497 +0x345
  google.golang.org/grpc.(*Server).processUnaryRPC()
      /go/src/google.golang.org/grpc/server.go:611 +0x10b6
  google.golang.org/grpc.(*Server).handleStream()
      /go/src/google.golang.org/grpc/server.go:769 +0xf9c
  google.golang.org/grpc.(*Server).serveStreams.func1.1()
      /go/src/google.golang.org/grpc/server.go:422 +0xb8

Goroutine 95 (running) created at:
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x73
  github.com/cockroachdb/cockroach/kv.grpcTransportFactory()
      /go/src/github.com/cockroachdb/cockroach/kv/transport_race.go:100 +0x2d8
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:981 +0x157
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:364 +0x27b
  github.com/cockroachdb/cockroach/kv.(*DistSender).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:250 +0x31d
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:459 +0x4da
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:344 +0x677
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:282 +0xbf
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:455 +0x6c8
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:344 +0x677
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:282 +0xbf
  github.com/cockroachdb/cockroach/kv.(*DistSender).getDescriptors()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:429 +0x101
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendChunk()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:672 +0x3d3
  github.com/cockroachdb/cockroach/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:591 +0x1a0
  github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).Send()
      /go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:383 +0x641
  github.com/cockroachdb/cockroach/internal/client.(*txnSender).Send()
      /go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:52 +0x120
  github.com/cockroachdb/cockroach/internal/client.(*DB).send()
      /go/src/github.com/cockroachdb/cockroach/internal/client/db.go:490 +0x2be
  github.com/cockroachdb/cockroach/internal/client.(*Txn).send()
      /go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:675 +0x675
  github.com/cockroachdb/cockroach/internal/client.(*Txn).(github.com/cockroachdb/cockroach/internal/client.send)-fm()
      /go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:359 +0x7c
  github.com/cockroachdb/cockroach/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/internal/client/db.go:417 +0x1ab
  github.com/cockroachdb/cockroach/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:359 +0xfd
  github.com/cockroachdb/cockroach/internal/client.runOneResult()
      /go/src/github.com/cockroachdb/cockroach/internal/client/db.go:517 +0x97
  github.com/cockroachdb/cockroach/internal/client.runOneRow()
      /go/src/github.com/cockroachdb/cockroach/internal/client/db.go:531 +0x6d
  github.com/cockroachdb/cockroach/internal/client.(*Txn).Get()
      /go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:221 +0xfa
  github.com/cockroachdb/cockroach/sql.(*LeaseManager).resolveName()
      /go/src/github.com/cockroachdb/cockroach/sql/lease.go:1061 +0xd1
  github.com/cockroachdb/cockroach/sql.(*LeaseManager).AcquireByName()
      /go/src/github.com/cockroachdb/cockroach/sql/lease.go:995 +0x138
  github.com/cockroachdb/cockroach/sql.(*planner).getTableLease()
      /go/src/github.com/cockroachdb/cockroach/sql/table.go:214 +0x4db
  github.com/cockroachdb/cockroach/sql.(*planner).(github.com/cockroachdb/cockroach/sql.getTableLease)-fm()
      /go/src/github.com/cockroachdb/cockroach/sql/scan.go:251 +0x4f
  github.com/cockroachdb/cockroach/sql.(*scanNode).initTable()
      /go/src/github.com/cockroachdb/cockroach/sql/scan.go:259 +0xe6
  github.com/cockroachdb/cockroach/sql.(*planner).getDataSource()
      /go/src/github.com/cockroachdb/cockroach/sql/data_source.go:243 +0x146e
  github.com/cockroachdb/cockroach/sql.(*planner).getDataSource()
      /go/src/github.com/cockroachdb/cockroach/sql/data_source.go:280 +0x178
  github.com/cockroachdb/cockroach/sql.(*planner).getSources()
      /go/src/github.com/cockroachdb/cockroach/sql/data_source.go:177 +0xae
  github.com/cockroachdb/cockroach/sql.(*selectNode).initFrom()
     /go/src/github.com/cockroachdb/cockroach/sql/select.go:438 +0x112
  github.com/cockroachdb/cockroach/sql.(*planner).SelectClause()
      /go/src/github.com/cockroachdb/cockroach/sql/select.go:295 +0x158
  github.com/cockroachdb/cockroach/sql.(*planner).Select()
      /go/src/github.com/cockroachdb/cockroach/sql/select.go:244 +0xa85
  github.com/cockroachdb/cockroach/sql.(*planner).prepare()
      /go/src/github.com/cockroachdb/cockroach/sql/plan.go:321 +0xe9b
  github.com/cockroachdb/cockroach/sql.(*Executor).Prepare()
      /go/src/github.com/cockroachdb/cockroach/sql/executor.go:366 +0x8c6
  github.com/cockroachdb/cockroach/sql.PreparedStatements.New()
      /go/src/github.com/cockroachdb/cockroach/sql/prepare.go:72 +0x84
  github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).handleParse()
      /go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:367 +0x689
  github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).serve()
      /go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:282 +0x8d1
  github.com/cockroachdb/cockroach/sql/pgwire.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/sql/pgwire/server.go:240 +0x989
  github.com/cockroachdb/cockroach/server.(*Server).Start.func8.1()
      /go/src/github.com/cockroachdb/cockroach/server/server.go:391 +0x88
  github.com/cockroachdb/cockroach/util/netutil.(*Server).ServeWith.func1()
      /go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:138 +0xc8

Goroutine 333 (finished) created at:
  google.golang.org/grpc.(*Server).serveStreams.func1()
      /go/src/google.golang.org/grpc/server.go:423 +0xba
  google.golang.org/grpc/transport.(*http2Server).operateHeaders()
      /go/src/google.golang.org/grpc/transport/http2_server.go:221 +0x126d
  google.golang.org/grpc/transport.(*http2Server).HandleStreams()
      /go/src/google.golang.org/grpc/transport/http2_server.go:282 +0x7f3
  google.golang.org/grpc.(*Server).serveStreams()
      /go/src/google.golang.org/grpc/server.go:424 +0x1d5
  google.golang.org/grpc.(*Server).serveNewHTTP2Transport()
      /go/src/google.golang.org/grpc/server.go:411 +0x4f0
  google.golang.org/grpc.(*Server).handleRawConn()
      /go/src/google.golang.org/grpc/server.go:388 +0x5c2
==================
```

Note to self: disabling the local call optimization in the stub test in #8890 lets this test fail 100% of the time with

>
--- FAIL: TestAdminAPICreateDatabase (10.51s)
	<autogenerated>:11: had 4 ranges at startup, expected 5

That can't be what's supposed to happen.

--- a/storage/store.go
+++ b/storage/store.go
@@ -2192,6 +2192,16 @@ func (s *Store) Send(ctx context.Context, ba roachpb.BatchRequest) (br *roachpb.
 			// after our operation started. This allows us to not have to
 			// restart for uncertainty as we come back and read.
 			h.Timestamp.Forward(now)
+			// We are going to hand the header (and thus the transaction proto)
+			// to the RPC framework, after which it must not be changed (since
+			// that could race). Since the subsequent execution of the original
+			// request might mutate the transaction, make a copy here.
+			//
+			// See #9130.
+			if h.Txn != nil {
+				clonedTxn := h.Txn.Clone()
+				h.Txn = &clonedTxn
+			}
 			pErr = s.intentResolver.processWriteIntentError(ctx, pErr, args, h, pushType)
 			// Preserve the error index.
 			pErr.Index = index

